src/Database/LSMTree/Normal.hs

```diff
@@ -315,7 +315,8 @@ rangeLookup = undefined
 -- Once a cursor has been created, updates to the referenced table don't affect
 -- the cursor.
 type Cursor :: (Type -> Type) -> Type -> Type -> Type -> Type
-data Cursor m k v blob
+-- TODO: tweak? needs Typeable?
+data Cursor m k v blob = forall h. Typeable h => Cursor !(Internal.Cursor m h)
```

Comment:
Same as for `TableHandle` and `BlobRef`. Do we currently use `Typeable` for anything there? Is the idea that we can later want to add runtime assertions?

Reply:
We're going to need it then, since reading entries from the cursor needs to wrap the `BlobRefs`. We can remove it in `Monoidal`, though, for TableHandle as well. I'll do that.